### PR TITLE
Support of optional members [19678]

### PR DESCRIPTION
**Pull request by richiware**
Also add support for collections of arrays.

Depends on:
- #65 
- eprosima/fastcdr#169

Related to:
- eprosima/fast-dds-gen#245

### CODE REVIEW

File: fastdds_python/test/types/CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [inline fastdds_python/test/types/CMakeLists.txt:1]: I have checked that regenerating the files using Fast DDS-Gen branch `feature/optional` generates the same files as in this PR. This CMakeLists.txt file differs.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: fastdds_python/test/types/CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [reply fastdds_python/test/types/CMakeLists.txt:1]: Whereas the test types seem to have been regenerated with the `feature/optional` branch of Fast DDS-Gen, the types used in the example package does not seem to.

Comment (richiware) [reply fastdds_python/test/types/CMakeLists.txt:1]: Done in cc32966

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**